### ISSUE TITLE
Validation results of ProGen is not on gold dataset?

### ISSUE THREAD

**Issue by JacksonWuxs** [HKUNLP/ProGen#2]
Hi authors,

I feel very interested in your series of works on data generation!
Currently, I am trying to develop my workflow for data generation based on your researchs (ProGen/ZeroGen/SunGen).

When I read through your code at main.py and cls_generation.py, I found that TAMs are evaluated on the synthesis validation set.
I am a little bit confused about this setting because I believe a fair comparison seems to be trained with the synthesis dataset but evaluated on the gold-valid dataset.
Otherwise, the TAM fine-tuned results on the ProGen/ZeroGen are not comparable with the prompting or traditional fine-tuned strategies.

Please correct me if I misunderstand your codes.

Best,
Xuansheng
